be/src/exec/pipeline/sink/hive_table_sink_operator.cpp

```diff
@@ -34,6 +34,9 @@ void HiveTableSinkOperator::close(RuntimeState* state) {
         if (!writer.second->closed()) {
             WARN_IF_ERROR(writer.second->close(state), "close writer failed");
         }
+        if (_rollback) {
+            WARN_IF_ERROR(writer.second->rollback(state), "writer rollback failed");
```

Comment:
Is  file elete operation underlying rollback a asynchronous one? if not so, it may be make the execute thread hang, so:
1.  can we put this operation into a special thread for processing?
2.  introduce pending_fnish status into HiveTableSinkOperator to make the driver pend util asynchronous file deletion finishes.

Reply:
It is synchronous but does not take too much time.

We could relax the constraints a bit, otherwise the logic will be much more complex.

Reply:
As long as it involves fe operations, it may get stuck. If close is stuck, it will cause the entire pipeline to get stuck.

Reply:
Fine. The time could be unpredictable if the network timeout.